### ISSUE TITLE
Runtime error when using UBSan with bfloat16 conversion intrinsic

### ISSUE THREAD

**Issue by ostannard** [llvm/llvm-project#61983]
When this code is compiled for AArch64 with UBSan, it reports a UBSan error, even though this is valid code:

```c
#include <arm_neon.h>

float foo(__bf16 a) {
  return vcvtah_f32_bf16(a);
}
```

```console
$ /work/llvm/build/bin/clang --target=aarch64--none-eabi -march=armv8.2-a+bf16 -c test.c -o - -S -O1 -fsanitize=undefined -fsanitize-minimal-runtime -fno-sanitize-recover
        .text
        .file   "test.c"
        .globl  foo                             // -- Begin function foo
        .p2align        2
        .type   foo,@function
foo:                                    // @foo
        .cfi_startproc
// %bb.0:                               // %entry
        stp     x29, x30, [sp, #-16]!           // 16-byte Folded Spill
        .cfi_def_cfa_offset 16
        mov     x29, sp
        .cfi_def_cfa w29, 16
        .cfi_offset w30, -8
        .cfi_offset w29, -16
        bl      __ubsan_handle_type_mismatch_minimal_abort
.Lfunc_end0:
        .size   foo, .Lfunc_end0-foo
        .cfi_endproc
                                        // -- End function
        .ident  "clang version 17.0.0 (git@github.com:llvm/llvm-project.git 626b7e5dd249f569203e024141c1a2a0f618df9c)"
        .section        ".note.GNU-stack","",@progbits
        .addrsig
```

**Comment by ostannard** [comment]
Fixed by #89062.

**Comment by ostannard** [comment]
Re-opening, this has been fixed for the original example, but still fails with an immediate value:
```c
#include <arm_neon.h>

int main() {
  vcvtah_f32_bf16(vcvth_bf16_f32(-0x1.aap4));

  return 0;
}
```

```
$ /work/llvm/build/bin/clang --target=aarch64-none-elf -march=armv8.2-a+bf16 -c file000103.c -o - -S -O1 -fsanitize=undefined -fsanitize-minimal-runtime -fno-sanitize-recover
clang: warning: argument unused during compilation: '-c' [-Wunused-command-line-argument]
        .text
        .file   "file000103.c"
        .globl  main                            // -- Begin function main
        .p2align        2
        .type   main,@function
        .word   3238382334                      // 0xc105cafe
        .word   2494312241                      // 0x94ac2f31
main:                                   // @main
// %bb.0:                               // %entry
        mov     w8, #-1043005440                // =0xc1d50000
        fmov    s0, w8
        bfcvt   h0, s0
        fmov    w8, s0
        tbnz    w8, #15, .LBB0_2
// %bb.1:                               // %vcvtah_f32_bf16.exit
        mov     w0, wzr
        ret
.LBB0_2:                                // %handler.shift_out_of_bounds.i
        stp     x29, x30, [sp, #-16]!           // 16-byte Folded Spill
        mov     x29, sp
        bl      __ubsan_handle_shift_out_of_bounds_minimal_abort
.Lfunc_end0:
        .size   main, .Lfunc_end0-main
                                        // -- End function
        .ident  "clang version 20.0.0git (git@github.com:llvm/llvm-project.git 1c984b86b389bbc71c8c2988d1d707e2f32878bd)"
        .section        ".note.GNU-stack","",@progbits
        .addrsig
```

That TBNZ instruction is branching to the UBSan handler if the sign bit of the bfloat result is set. The constant value is a valid, exact bfloat value: https://float.exposed/b0xc1d5
